spec/features/further_education_payments/happy_js_path_spec.rb

```diff
@@ -18,6 +18,7 @@
     expect(page).to have_content("Which FE provider are you employed by?")
     fill_in "Which FE provider are you employed by?", with: college.name
     within("#claim-provision-search-field__listbox") do
+      sleep(1)
```

Comment:
seems to help with the success rate of this test

Reply:
Try using `expect(page).to have_content("some content")` because Capybara has a built-in wait mechanism which eliminates the need for sleep calls.

Reply:
i'm aware of that but don't think it's the issue. as the line after this
```
find("li", text: college.name).click
```
should wait for the selector then click on it. this suggests that element appears first but the js event has yet to be bound. I've not investigated upstream yet to check if they add the element prior to the event being bound.

if this is the case it would make total sense as the element gets clicked but no js event handler is present so nothing happens instead of all the js that needs to happen

Reply:
I've had a look upstream but can't determine if there is a delay between when the element is added versus when the event handler is added. However adding the `sleep` clause on this specific line and not other locations seems to drastically improve success rate. which still suggests to me that the click always happens but its whether or not the js is fired